modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/main/scala/edu/gemini/seqexec/web/server/common/LogInitialization.scala

```diff
@@ -6,73 +6,77 @@ import java.util.logging.LogManager
 import scala.io.{Codec, Source}
 import scalaz._
 import Scalaz._
-import scalaz.effect.IO
+import scalaz.concurrent.Task
 
-trait LogInitialization {
-  sealed trait LogConfError
-  case class ConfigurationNotFound(s: String) extends LogConfError
-  case class LogDirNotFound(f: File) extends LogConfError
-  case object ConfigurationLoadingError extends LogConfError
+trait AppBaseDir {
+  /**
+    * Calculates the base dir of the application based on the location of "this" class jar file
+    * It will throw an exception if unable to find the base dir
+    */
+  def baseDir: Task[File] = Task.delay {
+    val clazz = this.getClass
+    val fileName = clazz.getResource(s"/${clazz.getName.replace(".", System.getProperty("file.separator"))}.class").getFile
 
-  type LogConf[A] = EitherT[IO, LogConfError, A]
```

Comment:
@tpolecat so you mean to say that `Task`s are always structured as a tree of threads, and when a `Task` throws an exception it can only be handled at the top level `Task`, when doing `unsafeXXX`, even for *async* exceptions (like a user interruption). Not being able to run *detached* threads rules out some concurrency patterns, but it also rules out many sources of complexity, specially when dealing with exceptions.

At a first glance, it seems that `.attempt` catches all exceptions indiscriminately which looks a bit [dangerous](https://hackage.haskell.org/package/base-4.9.0.0/docs/Control-Exception.html#g:4). But if it's only used at the top level and you make sure any unhandled exceptions are re-raised then I think it's quite reasonable to use it.

But I definitely have to read more about `Task`s because it's quite different to what I'm used to.